From a936718664dbfa17604ed34291a9a2a2c63c5058 Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Wed, 11 Jun 2025 16:37:10 +0300
Subject: [PATCH] [SW-228431] g3: sec: make tpc dbg gap write protected

This patch configures the TPC DBG gap regions to read only
any write to these regions not allowed and its violation.

Change-Id: I8cc95e1d9f334d464b9f0270d4c6e573ce2a3bdb
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
---

diff --git a/drivers/misc/hl_security/gaudi3/security.c b/drivers/misc/hl_security/gaudi3/security.c
index 45d9449..9c3489f 100644
--- a/drivers/misc/hl_security/gaudi3/security.c
+++ b/drivers/misc/hl_security/gaudi3/security.c
@@ -802,7 +802,7 @@
 		   val);
 	min = TPC_DBG_BASE + mmHD0_TPC0_EML_TPC_QM_BASE + HD0_TPC0_EML_TPC_QM_MAX_OFFSET;
 	max = TPC_DBG_BASE + mmHD0_TPC0_EML_CS_BASE - 1;
-	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_11, min, max, rr_sel, true, HL_SEC_RD_WR_PROT);
+	hl_secure_lbw_rr(dev, HL_SEC_LBW_RR_SHORT_11, min, max, rr_sel, true, HL_SEC_WR_PROT);
 
 	/* All PCIE type RRs(PCIE_ELBI, PCIE_MSTR, PCIE_LBW) for all RR instances are not configured
 	 * during security configurations. But clearing for all  should not have any adverse impact.
